common/src/main/java/com/brandwatch/minibcr/common/repository/query/QueryRepositoryImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.brandwatch.minibcr.common.repository.query;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import com.brandwatch.minibcr.common.domain.Query;
+
+@Repository
+public class QueryRepositoryImpl implements QueryRepository {
+
+    private static final String QUERY_TABLE = "queries";
+    private static final String ALL_FIELDS = "id, definition";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
```

Comment:
Loggers can be static, though you'll need to swap out `this.getClass()` for `QueryRepositoryImpl.class` as you can't do `this` in a static context. The above is fine with inherited classes though. Explanation [here](https://stackoverflow.com/questions/6653520/why-do-we-declare-loggers-static-final).